File: mesop/components/video/video.py
import mesop.components.video.video_pb2 as video_pb
from mesop.component_helpers import (
  Style,
  insert_component,
  register_native_component,
)


@register_native_component
def video(
  *,
  src: str,
  style: Style | None = None,
  key: str | None = None,
):
  """
  Creates a video.

  Args:
      src: URL of the video source
      style: The style to apply to the image, such as width and height.
  """
  insert_component(
    key=key,
    type_name="video",
    proto=video_pb.VideoType(
      src=src,
    ),
    style=style,
  )

File: mesop/components/video/video.ts
import {Component, Input} from '@angular/core';
import {
  Key,
  Style,
  Type,
} from 'mesop/mesop/protos/ui_jspb_proto_pb/mesop/protos/ui_pb';
import {VideoType} from 'mesop/mesop/components/video/video_jspb_proto_pb/mesop/components/video/video_pb';
import {formatStyle} from '../../web/src/utils/styles';

@Component({
  selector: 'mesop-video',
  templateUrl: 'video.ng.html',
  standalone: true,
})
export class VideoComponent {
  @Input({required: true}) type!: Type;
  @Input() key!: Key;
  @Input() style!: Style;
  private _config!: VideoType;

  ngOnChanges() {
    this._config = VideoType.deserializeBinary(
      this.type.getValue() as unknown as Uint8Array,
    );
  }

  config(): VideoType {
    return this._config;
  }

  getStyle(): string {
    return formatStyle(this.style);
  }
}
